### PR TITLE
fix problems in the spec

### LINKED ISSUE #193: [BUG:] Nullability error in the generated spec UserStatusStatus.clearAt
Yesterday I hit some corner cases within the spec.

The test effectively expects that the `UserStatusStatus.clearAt` is null although it isn't marked as nullable in the spec but still required.

https://github.com/provokateurin/nextcloud-neon/blob/e690ed2456ba508b5789a7a9f1a20b93aa221c17/packages/nextcloud/test/user_status.dart#L46-L48

https://github.com/provokateurin/nextcloud-neon/blob/e690ed2456ba508b5789a7a9f1a20b93aa221c17/packages/nextcloud/lib/src/nextcloud.openapi.json#L1774-L1835

The problem is that `UserStatusClearAt` does not have any requied fields either thus a valid serialized value would be `null` but `build_value` does not like the serialized object being null (and again the `UserStatusStatus.clearAt` is not marked as nullable).

https://github.com/google/built_value.dart/blob/5c71eee4a8890de2bbae85b798be1b9e8d171ce4/built_value/lib/serializer.dart#L336-L337
 
The screenshot shows the output of my vscode debugger.
Ideally the `UserStatusStatus.clearAt` should be marked as nullable. 

Screenshot_20230214_124421

Am I right with my suspicion that the spec is wrong?

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes #193 

as requested in https://github.com/provokateurin/nextcloud-neon/pull/194#issuecomment-1492937977

### CODE REVIEW

File: specs/core.json [hunk 1]
```diff
@@ -1311,4 +1317,4 @@
       }
     }
   }
-}
```
Comment (provokateurin) [inline specs/core.json:1314]: Newline

File: specs/notifications.json [hunk 2]
```diff
@@ -509,4 +499,4 @@
       }
     }
   }
-}
+}
```
Comment (provokateurin) [inline specs/notifications.json:502]: Newline

File: specs/user_status.json [hunk 3]
```diff
@@ -192,11 +191,7 @@
                 "$ref": "#/components/schemas/OCSMeta"
               },
               "data": {
-                "anyOf": [
-                  {
-                    "$ref": "#/components/schemas/Status"
-                  }
-                ]
+                "$ref": "#/components/schemas/Status"
```
Comment (provokateurin) [inline specs/user_status.json:194]: This might fail, since an empty array can be returned here if the user has no status. Maybe just go for the `[oneOf: [array, Status]]` thing?

File: specs/user_status.json [hunk 4]
```diff
@@ -245,11 +240,7 @@
                 "$ref": "#/components/schemas/OCSMeta"
               },
               "data": {
-                "anyOf": [
-                  {
-                    "$ref": "#/components/schemas/PublicStatus"
-                  }
-                ]
+                "$ref": "#/components/schemas/PublicStatus"
```
Comment (provokateurin) [inline specs/user_status.json:243]: Same as above

File: specs/user_status.json [hunk 5]
```diff
@@ -612,4 +601,4 @@
       }
     }
   }
-}
+}
```
Comment (provokateurin) [inline specs/user_status.json:604]: Newline

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: specs/core.json (hunk 1 above)
Comment (Leptopoda) [reply specs/core.json]: Bad formatter settings :)
